core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -555,6 +557,66 @@ void injectDependencies(BaseCommand baseCommand) {
         } while (!clazz.equals(BaseCommand.class));
     }
 
+    /**
+     * Instantiates a class with dependencies then registers a command with ACF
+     *
+     * @param commandClass the class of the command
+     */
+    public void registerCommand(Class<? extends BaseCommand> commandClass) {
+        registerCommand(constructCommand(commandClass));
+    }
+
+    /**
+     * Attempts to construct an instance of a command by injecting dependencies registered with
+     * {@link CommandManager#registerDependency(Class, Object)} into either the only constructor or the constructor
+     * annotated with {@link Dependency}.
+     *
+     * @param commandClass the class to instantiate
+     * @return the instance of the command
+     */
+    BaseCommand constructCommand(Class<? extends BaseCommand> commandClass) {
+        Constructor<?> constructor = null;
+        for (Constructor<?> possibleConstructor : commandClass.getConstructors()) {
+            if (annotations.hasAnnotation(possibleConstructor, Dependency.class)) {
+                constructor = possibleConstructor;
+                break;
+            }
+        }
+        if (commandClass.getConstructors().length == 1) {
+            constructor = commandClass.getConstructors()[0];
+        }
+        if (constructor == null) {
+            throw new IllegalStateException("Could not find suitable constructor for " + commandClass.getName());
+        }
+
+        Parameter[] parameters = constructor.getParameters();
+        Object[] args = new Object[parameters.length];
+        boolean accessible = constructor.isAccessible();
+        if (!accessible) {
+            constructor.setAccessible(true);
+        }
+        for (int i = 0; i < parameters.length; i++) {
+            String dependency = annotations.getAnnotationValue(parameters[i], Dependency.class);
+            String key = (key = dependency) == null || key.isEmpty() ? parameters[i].getType().getName() : key;
```

Comment:
I feel like this is much more understandable this way
```suggestion
            String key = dependency == null || dependency.isEmpty() ? parameters[i].getType().getName() : dependency;
```